src/components/PackageList/PackageList.tsx

```diff
@@ -0,0 +1,40 @@
+import { Center, Spinner } from "@chakra-ui/react";
+import { FunctionComponent } from "react";
+import { CatalogPackage } from "../../api/package/packages";
+import { PackageCardType } from "../PackageCard";
+import { CompactCardList } from "./CompactCardList";
+import { WideCardList } from "./WideCardList";
+
+const listViews = {
+  [PackageCardType.Compact]: CompactCardList,
+  [PackageCardType.Wide]: WideCardList,
+};
+
+export interface PackageListViewProps {
+  items: CatalogPackage[];
+}
+
+export interface PackageListProps extends Partial<PackageListViewProps> {
+  cardView?: PackageCardType;
+  loading?: boolean;
+  title?: string;
+}
```

Comment:
since `PackageListViewProps` only has 1 property, `Partial<PackageListViewProps>` is kinda funky but I assume we want that property optional  on `PackageListProps`?